File: XMLCheck/Handler/CheckRulesXML03/Xml3RuleRunner.cs
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML03
{
    public static class Xml3RuleRunner
    {
        private static readonly List<IXml3Rule> _rules = new()
        {

        };
        public static List<ErrorDetails> Run(List<Xml3Model>? models , HoSoContext context)
        {
            var errors = new List<ErrorDetails>();

            if (models == null || models.Count == 0)
                return errors;

            foreach (var model in models)
            {
                foreach (var rule in _rules)
                {
                    var err = rule.Check(model,context);
                    if (err != null)
                        errors.Add(err);
                }
            }

            return errors;
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml3/xml3RuleRunner.ts
import { Xml3Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { IXml3Rule } from './xml3RuleBase';

export class Xml3RuleRunner {
  private static readonly rules: IXml3Rule[] = [];

  public static run(models: Xml3Model[], context: HoSoContext): ErrorDetails[] {
    const errors: ErrorDetails[] = [];
    if (!models || models.length === 0) return errors;

    for (const model of models) {
      for (const rule of this.rules) {
        const err = rule.check(model, context);
        if (err) {
          errors.push(err);
        }
      }
    }

    return errors;
  }
}
